siad/main.go

```diff
@@ -111,8 +113,10 @@ func startEnvironment(*cobra.Command, []string) {
 	}
 
 	// join the network
+	//
+	// TODO: This should proabably happen in newDaemon.
 	if !config.Siacore.NoBootstrap {
-		go d.bootstrap()
+		go d.gateway.Bootstrap(modules.BootstrapPeers[0])
```

Comment:
perhaps. You'd have to add a `NoBootstrap` field to `daemonConfig`. And idk, I like that `newDaemon` just creates the daemon and nothing else. We're already calling `d.listen` in main anyway.